pkg/services/store/service.go

```diff
@@ -114,10 +117,39 @@ func ProvideService(sql *sqlstore.SQLStore, features featuremgmt.FeatureToggles,
 		return storages
 	}
 
-	return newStandardStorageService(sql, globalRoots, initializeOrgStorages)
+	authService := newStaticStorageAuthService(func(ctx context.Context, user *models.SignedInUser, storageName string) map[string]filestorage.PathFilter {
+		if user == nil || !user.IsGrafanaAdmin {
+			return nil
+		}
+
+		switch storageName {
+		case RootPublicStatic:
+			return map[string]filestorage.PathFilter{
+				ActionFilesRead:   allowAllPathFilter,
+				ActionFilesWrite:  denyAllPathFilter,
+				ActionFilesDelete: denyAllPathFilter,
+			}
+		case RootDevenv:
+			return map[string]filestorage.PathFilter{
+				ActionFilesRead:   allowAllPathFilter,
```

Comment:
this can be modified as needed - e.g.
```suggestion
				ActionFilesRead:   filestorage.NewPathFilter([]string{"/dev-dashboards/datasource-influxdb/"}, []string{"/dev-dashboards/"}, nil, nil),
```